pkg/platform/aws/aws.go

```diff
@@ -226,6 +226,35 @@ func (c *config) checkValidConfig() hcl.Diagnostics {
 
 	diagnostics = append(diagnostics, c.checkNotEmptyWorkers()...)
 	diagnostics = append(diagnostics, c.checkWorkerPoolNamesUnique()...)
+	diagnostics = append(diagnostics, c.checkNameSizes()...)
+
+	return diagnostics
+}
+
+// checkNameSizes checks the size of names since AWS has a limit of 32
+// characters on resources.
+func (c *config) checkNameSizes() hcl.Diagnostics {
+	var diagnostics hcl.Diagnostics
+
+	maxNameLen := 18 // 32 - len("-workers-https"). This is the longest resource suffix.
```

Comment:
Can we have this calculation in place rather than in the comment?

Reply:
Sure, I'll have to tame the linter though.